lib/asn1/gen.c

```diff
@@ -1963,9 +1988,22 @@ generate_type_header (const Symbol *s)
         fprintf(symsfile, "ASN1_SYM_TYPE(\"%s\", \"%s\", %s)\n",
                 s->name, s->gen_name, s->gen_name);
 
-    fprintf(headerfile, "typedef ");
-    define_type(0, s->gen_name, s->gen_name, NULL, s->type, TRUE,
-                preserve_type(s->name) ? TRUE : FALSE);
+    if (!s->emitted_declaration) {
+        fprintf(headerfile, "typedef ");
+        define_type(0, s->gen_name, s->gen_name, NULL, s->type,
+                    DEF_TYPE_TYPEDEFP | DEF_TYPE_EMIT_NAME |
+                    (preserve_type(s->name) ? DEF_TYPE_PRESERVE : 0));
+    } else if (s->type->type == TType) {
+        /* This is a type alias and we've already declared it */
```

Comment:
We've already emitted all the forward declarations, but since for type aliases the definition is the same as the declaration, we should just emit nothing here.

Reply:
This is the case of an alias of the form `Foo := Bar`.